docs/content/policy-language.md

```diff
@@ -1303,6 +1323,61 @@ behaviour of other rules.
 For using the `some` keyword with iteration, see
 [the documentation of the `in` operator](#membership-and-iteration-in).
 
+## Every Keyword
+
+```live:eg/every0:module:merge_down
+import future.keywords.every
+
+p {
+    every x in input.collection { x != "z" }
+}
+
+test_p {
+    x := p with input.collection as ["a", "r", "r", "a", "y"]
+    x == true
+}
+```
+```live:eg/every0:output
+```
+
+The `every` keyword takes an (optional) key argument, a value argument, a domain, and a
+block of further queries, its "body".
+
+The keyword is used to explicity assert that its body is true for *any element in the domain*.
+It will iterate over the domain, bind its variables, and check that the body holds
+for those bindings.
+If one of the bindings does not yield a successful evaluation of the body, the overall
+statement is undefined (and no further evaluation of the body occurs).
+
+If the domain is empty, the overall statement is true.
+
+Evaluating `every` does **not** introduce new bindings into the rule evaluation.
+
+Used with a key argument, the index, or property name (for objects), comes into the
+scope of the body evaluation:
+
+```live:eg/every1:module:merge_down
+import future.keywords.every
+
+p {
```

Comment:
I'd split this example in two... one for arrays and one for objects. I guess we'd want a third example for sets.